buildkite/scripts/cache.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -eou pipefail
+
+if [[ $# -ne 3 ]]; then
+  echo "Usage: $0 <save|restore> key path"
+  exit 1
+fi
+
+MODE=$1
+KEY=$2
+PATH=$3
+
+if [[ "$MODE" == "save" ]]; then
+  zip -r "$KEY.zip" "$PATH"
+  buildkite-agent artifact upload "$KEY.tar" "gs://buildkite/coda/shared"
```

Comment:
pretty confusing but the path should be specified in an EnvVar (BUILDKITE_ARTIFACT_UPLOAD_DESTINATION) and the script can just set the upload destination as relative to that (in this case, something like `$KEY.tar`).